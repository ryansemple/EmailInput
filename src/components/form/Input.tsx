import React, { ChangeEvent } from "react";

interface InputProps {
	className: string,
	value: string,
	onChange: (value: string) => void
}

const Input = (props: InputProps) => 
{
	const onChange = (event: ChangeEvent<HTMLInputElement>): void => 
	{
		const value: string = event.currentTarget.value;
		props.onChange(value);
	}

	return (
		<input 
			className={`form_input ${props.className}`}
			value={props.value}
			onChange={onChange}
		/>
	)
}

export default Input;