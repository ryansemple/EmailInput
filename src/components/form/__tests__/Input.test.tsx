import React from "react";
import { render } from "@testing-library/react";
import Input from "../Input";

test("Renders Button component correctly.", () => {
	const { container } = render
	(
		<Input
			className="class"
			value="abc"
			onChange={(value: string) => { alert("changed") }}
			name="input"
		/>
	);
	expect(container).toMatchSnapshot();
});