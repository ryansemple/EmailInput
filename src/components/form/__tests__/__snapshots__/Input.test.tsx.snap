// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Button component correctly. 1`] = `
<div>
  <input
    class="form_input class"
    name="input"
    value="abc"
  />
</div>
`;
